### PR TITLE
STR-13 Setup Login

### PR DESCRIPTION
**Pull request by mbendandour**
# Description
When given a cookie (username/email and password), what will be returned is the user id of a valid user, whether it is valid (denoted by 1 for yes, 0 for no), and a unique token.
  
# Testing
run phpunit

**Using Postman**

- Make sure you select post, and that the api route is {your localhost}/loginVerification

- Click on _Body_

- For 'Key' enter: username
                             password

- For 'Value' enter: steve@csun.edu (for username)
  Password is left blank

- Hit Send

- Should receive {"user_id":" ","valid":"1","token":" "} structure with information. 

# Installation
Do a composer install

### CODE REVIEW

File: app/Services/AuthenticationService.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+namespace App\Services;
+
+
+use App\Contracts\AuthenticationContract;
+
+
+class AuthenticationService implements AuthenticationContract
+{
+
+    public function __construct()
```
Comment (liadGolan) [inline app/Services/AuthenticationService.php]: If nothing is needed in the constructor, it does not to be in here

**Review by liadGolan — CHANGES_REQUESTED** [review]
Please delete the constructor and this still needs a service test

File: app/Http/Controllers/LoginController.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+        $credentials = [
+            'username'=> $request->username,
+            'password'=>$request->password
+        ];
+
+        $hashedKey = $this->createHashedKey();
+        $token = $this->createNewToken($hashedKey);
+        $userInformation = $this->authenticationUtility->authenticateUser($credentials);
+
```
Comment (frlzjosh) [inline app/Http/Controllers/LoginController.php:56]: suggestion: To make more understandable would you be open to making this variable name `userInformationFromToken` ?

File: app/Models/User.php [hunk 3]
```diff
@@ -2,21 +2,23 @@
...
+class User extends MetaUser
 {
     use Notifiable;
 
+    public $incrementing = false;
     /**
      * The attributes that are mass assignable.
      *
-     * @var array
+     * @var array primary string thatis a string eloquent
```
Comment (frlzjosh) [inline app/Models/User.php]: really small, a space is missing in `thatis`

File: app/Models/User.php [hunk 4]
```diff
@@ -2,21 +2,23 @@
...
      */
     protected $fillable = [
-        'name', 'email', 'password',
+        'user_id','username', 'email', 'password',
```
Comment (frlzjosh) [inline app/Models/User.php]: hm... the PSR_2 should've fixed spacing here

File: routes/web.php [hunk 5]
```diff
@@ -15,7 +15,4 @@
     return view('welcome');
 });
 
-Route::get('/get_role/{email}', 'UserRoleController@getRole');
-
-/** Allows laravel to be able to serve/host any vue page */
-Route::get('/{any}', 'SpaController@index')->where('any', '.*');
+Route::post('loginVerification', 'LoginController@authenticateUser');
```
Comment (frlzjosh) [inline routes/web.php]: you need to add back the routes on line 18 and 21. `SpaController` is to return vue pages and `UserRoleController` is so front-end can hit that api

File: routes/web.php [hunk 6]
```diff
@@ -15,7 +15,5 @@
...
+Route::get('logout','LoginController@logout');
```
Comment (frlzjosh) [inline routes/web.php]: You need to add back the `get_role` and `/{any}' routes. They are needed for front-end to hit the api and so Laravel can load vue pages properly

**Review by frlzjosh — CHANGES_REQUESTED** [review]
Really Minor changes: A few things need to be added back and it seems as if you do not have the PSR_2 pre commit on your project yet

File: app/Http/Controllers/LoginController.php (hunk 2 above)
Comment (mbendandour) [reply app/Http/Controllers/LoginController.php:56]: Yes

**Review by liadGolan — APPROVED** [review]
👍

File: routes/web.php (hunk 6 above)
Comment (liadGolan) [reply routes/web.php]: Not needed for front end, but it is needed for testing api's on our end

**Review by nercesk — APPROVED** [review]

File: app/Models/User.php (hunk 4 above)
Comment (mbendandour) [reply app/Models/User.php]: ....does that mean I don't have PSR_2?

Comment (frlzjosh) [reply app/Models/User.php]: Just talked to Liad, it seems as if you do!

**Review by frlzjosh — APPROVED** [review]
Nice!

**Pull request merged**